lib/shared-sku-app/src/SkuPicker/SkuPicker.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { Component } from 'react';
+import get from 'lodash/get';
+import clamp from 'lodash/clamp';
+import debounce from 'lodash/debounce';
+import { Button, TextInput, Icon } from '@contentful/forma-36-react-components';
+import { AppExtensionSDK } from 'contentful-ui-extensions-sdk';
+import { Divider } from '../Divider';
+import { ProductList } from './ProductList';
+import { Paginator } from './Paginator';
+import { Product } from '../interfaces';
+import { Pagination } from './interfaces';
+import { ProductSelectionList } from './ProductSelectionList';
+import { styles } from './styles';
+
+export interface Props {
+  sdk: AppExtensionSDK;
+  fetchProductPreview: Function;
+  fetchProducts: Function;
+}
+
+interface State {
+  activePage: number;
+  search: string;
+  pagination: Pagination;
+  products: Product[];
+  selectedProducts: Product[];
+  selectedSKUs: string[];
+}
+
+const SEARCH_DELAY = 250;
+
+function getSaveBtnText(selectedSKUs: string | string[]): string {
+  if (typeof selectedSKUs === 'string') {
+    return 'Save product';
+  }
+
+  switch (selectedSKUs.length) {
+    case 0:
+      return 'Save products';
+    case 1:
+      return 'Save 1 product';
+    default:
+      return `Save ${selectedSKUs.length} products`;
+  }
+}
+
+export class SkuPicker extends Component<Props, State> {
+  state: State = {
+    activePage: 1,
+    search: '',
+    pagination: {
+      count: 0,
+      limit: 0,
+      offset: 0,
+      total: 0
+    },
+    products: [],
+    selectedProducts: [],
+    selectedSKUs: get(this.props, ['sdk', 'parameters', 'invocation', 'fieldValue'], [])
+  };
+
+  async componentDidMount() {
+    await this.updateProducts();
+    await this.updateSelectedProducts();
+  }
+
+  setSearchCallback = debounce(() => {
+    this.setActivePage(1);
+    this.updateProducts();
+  }, SEARCH_DELAY);
+
+  setSearch = (search: string) => {
+    this.setState({ search }, this.setSearchCallback);
+  };
+
+  updateProducts = async () => {
+    try {
+      const {
+        activePage,
+        pagination: { limit },
+        search
+      } = this.state;
+      const offset = (activePage - 1) * limit;
+      const { pagination, products } = await this.props.fetchProducts(search, { offset });
+      this.setState({ pagination, products });
+    } catch (error) {
+      this.props.sdk.notifier.error('There was an error fetching the product list.');
+    }
+  };
+
+  updateSelectedProducts = async () => {
+    try {
+      const selectedProductsPromises = this.state.selectedSKUs.map(sku =>
```

Comment:
Does it mean we do 1 HTTP request per SKU stored? Could we use some kind of collection endpoint?

Reply:
Yup this was WIP, now it's updated :)